server/src/main/java/org/elasticsearch/action/admin/cluster/stats/AnalysisStats.java

```diff
@@ -176,6 +196,34 @@ private static void aggregateAnalysisTypes(
         }
     }
 
+    private static void aggregateSynonymsSetStats(
+        Collection<Settings> filterSettings,
+        Map<String, SynonymsStats> synonymsStats,
+        String indexName,
+        Set<String> synonymsIdsUsed,
+        Set<String> synonymIdsUsedInIndices
+    ) {
+        for (Settings filterComponentSettings : filterSettings) {
+            final String type = filterComponentSettings.get("type");
+            if (SYNONYM_FILTER_TYPES.contains(type)) {
+                for (String synonymRuleType : SYNONYM_RULES_TYPES) {
+                    String synonymId = filterComponentSettings.get(synonymRuleType);
+                    if (synonymId != null) {
+                        boolean isInline = "synonyms".equals(synonymRuleType);
```

Comment:
Nitpick: Stylistically would it be better to compare the `synonymRuleType` against a variable String?

Reply:
I'm not sure what you mean... I was thinking on extracting "synonyms" as a constant, but didn't see that pattern used much in here. What would be your preferred code here?

Reply:
I don't have a strong opinion, it's OK the way it is. I just sometimes think the pattern "foo".equals(x) looks odd 😂

Reply:
> "foo".equals(x) looks odd

Totally! But it protects from potential NPEs - that's why I adopted it (and Checkstyle pounded me so much in the past that it's a reflex now) 😅

Reply:
My concern is for this line `filterComponentSettings.get(synonymRuleType)`  when synonymRuleType = "synonyms".  As this string could be huge, it may take some time, and we want cluster stats to be very fast. (May be my concert is overrated, but still better to think of alternative implementation).

May be, an alternative could be to iterate over `SYNONYM_RULES_TYPES`, check if  `filterComponentSettings.hasValue` for this key, if the key is "synonyms" don't retrieve it.

Reply:
I understand your concern. This small piece of code has been through many iterations from my side, as I understand the need to make it as fast as possible.

I think that `get` vs `containsKey` should be negligible - we're not copying the String value to some other variable, but just returning the String associated in the map (the reference to it, and not reading it afterwards). I can't find specific performance differences on it when searching around - can you provide one for my education?

TBH, I am more concerned about `"synonyms".equals(synonymRuleType)` as this does String comparison. I was thinking on changing it for the following:

Move the inline type first:
`    static final String[] SYNONYM_RULES_TYPES = { "synonyms", "synonyms_set", "synonyms_path" };` 

Take advantage of knowing that the first rule type is the inline one:

```java
                boolean isInline = true;
                for (String synonymRuleType : SYNONYM_RULES_TYPES) {
                    String synonymId = filterComponentSettings.get(synonymRuleType);
                    if (synonymId != null) {
                        SynonymsStats stat = synonymsStats.computeIfAbsent(synonymRuleType, id -> new SynonymsStats());
                        if (synonymIdsUsedInIndices.add(synonymRuleType + indexName)) {
                            stat.indexCount++;
                        }
                        if (isInline || synonymsIdsUsed.add(synonymRuleType + synonymId)) {
                            stat.count++;
                        }
                        break;
                    }
                    isInline = false;
                }
```

What do you think?

Reply:
Thanks for explaining Carlos! A potential problem with `filterComponentSettings.get("synonyms")` that it needs to covert list to String.

```java
            if (SYNONYM_FILTER_TYPES.contains(type)) {
                boolean isInline = false;
                String synonymRuleType = "synonyms_set";
                String synonymId = filterComponentSettings.get(synonymRuleType);
                if (synonymId == null) {
                    synonymRuleType = "synonyms_path";
                    synonymId = filterComponentSettings.get(synonymRuleType);
                }
                if (synonymId == null) {
                    synonymRuleType = "synonyms";
                    isInline = true;
                }
                SynonymsStats stat = synonymsStats.computeIfAbsent(synonymRuleType, id -> new SynonymsStats());
                if (synonymIdsUsedInIndices.add(synonymRuleType + indexName)) {
                    stat.indexCount++;
                }
                if (isInline || synonymsIdsUsed.add(synonymRuleType + synonymId)) {
                    stat.count++;
                }
            }
```

Reply:
You're totally right! I thought I was dealing with a `Map` instance, but it's a `Settings` one 🤦 . Thanks for your feedback!

I'm adding your solution - I can't think of anything more expressive than that. Thanks!

Reply:
Thanks Carlos, it is your solution, slightly modified.